### PR TITLE
Add video (squashed)

### LINKED ISSUE #267: adding video support
before I get too deep into coding this up, I want to solicit input about where to add the code and how to organize. My initial thinking is:

- a `Movie` class in [_visual.py](https://github.com/LABSN/expyfun/blob/master/expyfun/visual/_visual.py) with methods play, stop, get_volume, set_volume
- various public and private methods/properties added to `ExperimentController`, such as:
    - `ec.player` (accessing `pyglet.media.Player` instance, analogous to `ec.window`)
    - `ec.load_movie()`
    - etc

Some questions / possible API changes:
- things like `ec.play()`, `ec.playing` etc: should they be changed to `ec.play_audio()` and `ec.audio_playing`, for parallelism with the to-be-added `ec.play_video` and `ec.video_playing`?
- what to do about the audio streams associated with loaded video files?  My current use case doesn't need them (I explicitly removed them when encoding, in fact) so for now could be ignored as YAGNI?
- dealing with video frame rate vs `ec.flip()`.  This sounds like a nightmare.  A possible easy solution for now would be to disable drawing and flipping while `ec.video_playing` is True.

Thoughts?

### PR DESCRIPTION
**Pull request by drammock**
Closes #268. Closes #267.

Ready for review @Eric89GXL @rkmaddox.  Please be sure to `make test` locally; pyglet is something of a moving target so I don't want any surprises due to different pyglet versions.

### CODE REVIEW

**Comment by drammock** [comment]
Travis is choking on the video test (`WAVEFormatException: AVbin is required to decode compressed media`).  @Eric89GXL you're the travis-master, how should I handle?  Installing AVbin is done via executable binary [here](http://avbin.github.io/AVbin/Download.html) (currently v10) or from linux repos as libavbin0 (currently version 7).  Should I test video only when `DEPS="full"`?

AppVeyor, on the other hand, is choking on `test_logging`?

**Comment by Eric89GXL** [comment]
I'd try the executable version it's more likely to work.

I'll try to take a look at the logging test soon, just ignore it for now

File: .travis.yml [hunk 1]
```diff
@@ -32,6 +32,10 @@ before_install:
         conda install --yes scipy matplotlib nose coverage setuptools;
         travis_retry pip install -q coveralls;
       fi;
+    # install AVbin
+    - wget https://github.com/downloads/AVbin/AVbin/install-avbin-linux-x86-64-v10 -O avbin.sh
```
Comment (drammock) [inline .travis.yml]: @Eric89GXL any idea why this `wget` command returns "4" (network error) all 3 times that travis tried it, but works fine locally?

Comment (Eric89GXL) [inline .travis.yml]: The error is `Unable to establish SSL connection`, which suggests that it's an SSL protocol failure. Either a bug in Ubuntu/wget or GitHub's certs. Can you use http?

Comment (drammock) [inline .travis.yml]: Thanks, I'll look into that. How did you figure out it was ssh? All I saw
in the logs was "error code 4" which is supposedly a generic network error.
How could I have dug deeper?

**Comment by Eric89GXL** [comment]
I looked here:

https://travis-ci.org/LABSN/expyfun/jobs/112970700#L279

What logs did you look at?

**Comment by drammock** [comment]
Same logs. Apparently I was blind to all text that was not highlighted red.
Sorry.

**Comment by rkmaddox** [comment]
Tested on my machine. No errors in `noestests`. One warning on every play:

```python
/home/rkmaddox/anaconda2/lib/python2.7/site-packages/pyglet/media/__init__.py:1170: DeprecationWarning: Player.eos_action is deprecated in favor of SourceGroup.loop and SourceGroup.advance_after_eos
  category=DeprecationWarning)
```

Playback was choppy on my system, compared to playback in mplayer, so it appears it was missing several frames. The good news is that playback length was unaffected, so it seems to sticking to master clock, rather than just playing the next frame in the sequence regardless of when the previous frame was.

**Comment by drammock** [comment]
thanks for pointing out the choppiness, @rkmaddox.  I'll try running on
MARLO to see if this is just the "linux halves the frame rate" issue, or
something more insidious.

**Comment by rkmaddox** [comment]
The choppiness did seem regular, so it could be a half (or quarter?) framerate problem.

**Comment by drammock** [comment]
we'll see if the CIs are happy now.  @rkmaddox that warning you mentioned is coming from `pyglet.__init__` which apparently didn't get the memo about `eos_action` being deprecated (FWIW, it looks like the replacement is not really fully implemented yet either, so for now we have to survive without end-of-stream events getting issued).

**Comment by drammock** [comment]
After some digging: the travis failures are probably due to an old bug in `wget`.  We're running the tests on 12.04 (precise) which ships with wget version 1.13.4-2ubuntu1.2.  Apparently 14.04 (trusty) ships with a version of `wget` that has the bug fixed (1.15-1ubuntu1.14.04.1).  So options are:

1. run our tests on trusty instead of precise
1.  pass `--no-check-certificate` to `wget`

@Eric89GXL do you have a preference between those two options?  Note that `apt-get install wget` will not work, since no newer version of `wget` was ever released in precise-updates or precise-backports.

**Comment by Eric89GXL** [comment]
I didn't know Travis had an OS version option. Does it?

I'm fine with the no check option

**Comment by drammock** [comment]
you can specify it in `.travis.yml`:
https://docs.travis-ci.com/user/trusty-ci-environment/

**Comment by Eric89GXL** [comment]
Didn't know that -- I wonder when they rolled it out. Would have saved me a
lot of pain some months ago having to deal with old 12.04 junk...

Anyway, +1 to try the upgrade.

**Comment by drammock** [comment]
here goes nothin'!

**Pull request closed without merging**